common/check_transfer_queues_status

```diff
@@ -1,85 +1,72 @@
- WHEN state = 'A' THEN 'queues.requests.submissionfailed.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'U' THEN 'queues.requests.suspend.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- WHEN state = 'P' THEN 'queues.requests.preparing.' || replace(activity, ' ', '_') || '.' || nvl(replace(substr(external_host, 9, length(external_host)-13), '.', '_'), 'no_fts_host')
- ELSE state
-END state_desc,
-num_rows
-FROM
-(
-select state, count(*) num_rows, activity, external_host
-FROM {schema}requests
-GROUP BY state, activity, external_host
-)""".format(schema=schema)
-
-PROM_SERVERS = config_get('monitor', 'prometheus_servers', raise_exception=False, default='')
-if PROM_SERVERS != '':
-    PROM_SERVERS = PROM_SERVERS.split(',')
-
 if __name__ == "__main__":
     try:
-        registry = CollectorRegistry()
-        g = Gauge('conveyor_queues_requests', '', labelnames=('state', 'activity', 'external_host'), registry=registry)
         session = get_session()
-        for k in session.execute(active_queue).fetchall():
-            print(k[0], k[1], end=" ")
-            probe_metrics.gauge(name=k[0].replace('-', '_')).set(k[1])
-            items = k[0].split('.')
-            state = items[2]
-            activity = items[3]
-            external_host = items[4].replace('-', '_')
-            g.labels(**{'activity': activity, 'state': state, 'external_host': external_host}).set(k[1])
-        if len(PROM_SERVERS):
-            for server in PROM_SERVERS:
-                try:
-                    push_to_gateway(server.strip(), job='check_transfer_queues_status', registry=registry)
-                except:
-                    continue
-    except:
+        statement = select(
+            func.count(),
+            models.Request.state,
+            models.Request.activity,
+            models.Request.external_host
+        ).group_by(
+            models.Request.state,
+            models.Request.activity,
+            models.Request.external_host
+        )
+
+        with PrometheusPusher() as manager:
+            for row in session.execute(statement):
+                activity = row.activity.replace(" ", "_")
+
+                # host 1.2.3.45 -> 1_2_3_45, or "no_fts_host" if there is no host
+                if row.external_host is not None:
+                    start = 9
+                    stop = len(row.external_host) - 13
+                    if stop < start: 
+                        external_host = 'no_fts_host'
+                    else:
+                        external_host = row.external_host[start, stop].replace(".", "_")
+                else:
+                    external_host = 'no_fts_host'
```

Comment:
The original query contained this ugliness due to necessity, but there’s no excuse to copy it over. It relies on the external host beginning with `https://` end ending with a four-digit port (typically `:8446`). Please use [`urlparse()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) instead.